sw/airborne/filters/median_filter.h

```diff
@@ -75,61 +76,175 @@ inline int32_t update_median_filter(struct MedianFilterInt *filter, int32_t new_
     }
     filter->sortData[j + 1] = temp;
   }
-  return filter->sortData[(MEDIAN_DATASIZE) >> 1]; // return data value in middle of sorted array
+  // return data value in middle of sorted array
+  return get_median_filter_i(filter);
 }
 
-inline int32_t get_median_filter(struct MedianFilterInt *filter)
+inline int32_t get_median_filter_i(struct MedianFilterInt *filter)
 {
-  return filter->sortData[(MEDIAN_DATASIZE) >> 1];
+  if (filter->size % 2 == 0){
+    return (filter->sortData[filter->size / 2] + filter->sortData[filter->size / 2 - 1]) / 2;
```

Comment:
I'm not completely happy this, because it goes again one of the property of the median filter which is that the output value is always one of the input elements. It is not the case here anymore. Not sure it is the most useful property, but still. I can see two options here:
- the size parameter is always set to the closest upper odd value
- the size parameter is only the "half-size", so the final size is (2*size+1)